orbit/cmd/orbit/orbit.go

```diff
@@ -297,16 +297,32 @@ func main() {
 						if err = keystore.AddSecret(secret); err != nil {
 							log.Warn().Err(err).Msgf("failed to add enroll secret to %v", keystore.Name())
 						} else {
-							log.Info().Msgf("added enroll secret to keystore: %v", keystore.Name())
-							deleteSecretPathIfExists(enrollSecretPath)
+							// Sanity check that the secret was added to the keystore.
+							checkSecret, err := keystore.GetSecret()
+							if err != nil {
+								log.Warn().Err(err).Msgf("failed to check that enroll secret was saved in %v", keystore.Name())
+							} else if checkSecret != secret {
+								log.Warn().Msgf("enroll secret was not saved correctly in %v", keystore.Name())
```

Comment:
In this case should we remove it from the keychain?
Or is the fact that the file exists enough to get the secret from the file?

Reply:
orbit will read from the file first and use that (that is what admins can use to rotate secrets)

Reply:
So in this case it will overwrite what was written in the first attempt?